### PR TITLE
Add Shai testing harness

### LINKED ISSUE #1848: Add testing harness for Shai
Factored out of #1113

In order write integration tests for Shai, we'll need a way of exercising the server. We'll eventually want something that can serve as a client for at least a few high-level integration tests, but the most expedient and principled way to move forward at this point is to set up a test harness for unit tests (mocked, as needed).

## Actions

- [x] Launch server via `server` subcommand
- [x] Add zio test dependencies
- [x] Set up simple test to obtain a connection

The client will expose a JSON interace.

### PR DESCRIPTION
**Pull request by shonfeder**
Closes #1848

This adds a trivial test to setup the dependencies and basic scaffolding for unit tests for the server. Most of the work involved in preparing this change was just in figuring out exactly what we should be testing at this stage, and troubleshooting some tricky dependencies and build configuration.

It turns out that the right thing to test at this point is just the core logic of the gRPC service. This will let us exercise the logic exposed by gRCP without also testing the gRPC library itself. Once we have non-trivial functionality in place, we can begin considering what kinds of integration tests we want to add.

- [x] Tests added for any new code
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [x] Documentation added for any new functionality

[unreleased]: https://github.com/informalsystems/apalache/tree/unstable/.unreleased

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -183,8 +183,27 @@ lazy val tla_bmcmt = (project in file("tla-bmcmt"))
       libraryDependencies += Deps.scalaCollectionContrib,
   )
 
+lazy val shai = (project in file("shai"))
```
Comment (shonfeder) [inline build.sbt:186]: Correction location in in file (should be prior to tool, since tool depends on it) and name of project.

Comment (thpani) [reply build.sbt:186]: I find it oddly gratifying that this project is (partially) called like the Fremen word for sandworm 😅

File: mod-tool/src/main/scala/at/forsyte/apalache/tla/Tool.scala [hunk 2]
```diff
@@ -345,13 +346,14 @@ object Tool extends LazyLogging {
   }
 
   private def runServer(executor: PassChainExecutor, server: ServerCmd): Int = {
-    logger.info("Running server...")
+    logger.info("Starting server...")
 
     // NOTE Must go after all other options are set due to side-effecting
     // behavior of current OutmputManager configuration
```
Comment (thpani) [inline mod-tool/src/main/scala/at/forsyte/apalache/tla/Tool.scala:352]:
```suggestion
    // behavior of current OutputManager configuration
```

**Review by thpani — APPROVED** [review]
👍

**Comment by shonfeder** [comment]
I hadn’t thought of the connection [Shia-Hulud](https://dune.fandom.com/wiki/Shai-Hulud) :D Nice one!

Thanks for the review.

**Pull request merged**